zigzag/zigzag.py

```diff
@@ -13,259 +13,274 @@
+        """Construct a qTest swagger model for all the JUnitXML test cases.
 
-    Returns:
-        list(str): An ordered list of strings to use for the qTest results hierarchy.
+        Returns:
+            list(AutomationTestLogResource): A list of qTest swagger model test logs.
+        """
 
-    Raises:
-        KeyError: missing test suite property.
-        AttributeError: the testcase 'classname' attribute is invalid
-    """
+        serialized_junit_xml = etree.tostring(self._junit_xml, encoding='UTF-8', xml_declaration=True)
+        testsuite_props = {p.attrib['name']: p.attrib['value'] for p in self._junit_xml.findall('./properties/property')}  # noqa
+        date_time_now = datetime.utcnow()
+        test_logs = []
 
-    module_hierarchy = [testsuite_props['RPC_RELEASE'],             # RPC Release Version (e.g. 16.0.0)
-                        testsuite_props['JOB_NAME'],                # CI Job name (e.g. PM_rpc-openstack-pike-xenial_mnaio_no_artifacts-swift-system) # noqa
-                        testsuite_props['MOLECULE_TEST_REPO'],      # (e.g. molecule-validate-neutron-deploy)
-                        testsuite_props['MOLECULE_SCENARIO_NAME']]  # (e.g. "default")
+        for testcase_xml in self._junit_xml.findall('testcase'):
+            testcase_status = 'PASSED'
 
-    testcase_groups = TESTCASE_GROUP_RGX.search(testcase_xml.attrib['classname']).groups()
+            test_log = swagger_client.AutomationTestLogResource()
+            test_log.properties = []
 
-    module_hierarchy.append(testcase_groups[0])         # Always append at least the filename of the test grouping.
-    if testcase_groups[1]:
-        module_hierarchy.append(testcase_groups[1])     # Append the class name of tests if specified.
+            if testcase_xml.find('failure') is not None or testcase_xml.find('error') is not None:
+                testcase_status = 'FAILED'
 
-    return module_hierarchy
+                possible_messages = [testcase_xml.find('error'), testcase_xml.find('failure')]
+                message = "\n".join([x.attrib['message'] for x in possible_messages if x is not None])
 
+                test_log.properties.append(swagger_client.PropertyResource(field_id=self._failure_output_field_id, field_value=message))  # noqa
 
-def _get_testsuite_props(junit_xml):
-    """Get a dictionary of testsuite properties from the JUnitXML results file.
+            elif testcase_xml.find('skipped') is not None:
+                testcase_status = 'SKIPPED'
 
-    Args:
-        junit_xml (ElementTree): A XML element representing a JUnit style testsuite result.
+            try:
+                test_log.name = self._TESTCASE_NAME_RGX.match(testcase_xml.attrib['name']).group(1)
+                test_log.automation_content = testcase_xml.find("./properties/property/[@name='test_id']").attrib['value']  # noqa
+                test_log.exe_start_date = testcase_xml.find("./properties/property/[@name='start_time']").attrib['value']  # noqa
```

Comment:
You can avoid the `# noqa` mark by utilizing a line continuation.

```python
test_log.exe_start_date = \
                    testcase_xml.find("./properties/property/[@name='start_time']").attrib['value']
```

Reply:
fixed